### PR TITLE
Code health improvements

### PR DESCRIPTION
**Pull request by nikhilwoodruff**
Adds:
* A test for lower-case variable names
* Group encapsulation
* (also, a test for entity_variable_name names, but I've disabled for now due to the tax variables)

Applies autopep8, black and linecheck.

### CODE REVIEW

File: openfisca_us/tests/code_health/variable_names.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+    system = CountryTaxBenefitSystem()
+    for instance in system.variables.values():
+        variable = type(instance)
+        assert (
+            variable.__name__.islower()
+        ), f"{variable.__name__} is not all lowercase"
+        if variable.entity.key in GROUP_ENTITY_NAMES:
+            pass
+            # assert variable.__name__[:len(variable.entity.key)] == variable.entity.key, f"{variable.__name__} does not begin with the specified group entity"
```
Comment (MaxGhenis) [inline openfisca_us/tests/code_health/variable_names.py:20]: Still a draft?

File: openfisca_us/entities.py [hunk 2]
```diff
@@ -32,6 +15,7 @@
             doc="A member of the household",
         ),
     ],
+    containing_entities=["spm_unit"],
```
Comment (MaxGhenis) [inline openfisca_us/entities.py:18]: I don't think the set of `containing_entities` arguments here consistently represents the hierarchy:

`person` -> `tax_unit` -> `family` -> `spm_unit` -> `household`

Should this include just the immediate predecessor in the hierarchy? SPMUnit having both family and household seems off.

**Review by MaxGhenis — COMMENTED** [review]
Thanks @nikhilwoodruff this will make our lives much easier!

File: openfisca_us/tests/code_health/variable_names.py (hunk 1 above)
Comment (nikhilwoodruff) [reply openfisca_us/tests/code_health/variable_names.py:20]: Yeah, the taxcalc variables in particular. We just need to add `tax_unit` to each variable name, but not sure how that'll work with the non-dep sums.

File: openfisca_us/entities.py (hunk 2 above)
Comment (nikhilwoodruff) [reply openfisca_us/entities.py:18]: I mean, I should know which is the correct encapsulation, but in https://github.com/PolicyEngine/openfisca-us/issues/233 `spm_unit` and `family` are switched.

Comment (MaxGhenis) [reply openfisca_us/entities.py:18]: Sorry - yes #233 was wrong (edited to fix). See https://www.census.gov/content/dam/Census/library/publications/2021/demo/p60-275.pdf#page=3

image

**Pull request merged**